ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -385,11 +398,41 @@ def uplink_switches(self):
 
     @cached_property
     def uplink_interfaces(self):
-        return get(self._switch_data_combined, "uplink_interfaces")
+        return self._range_expand(default(
+            get(self._switch_data_combined, "uplink_interfaces"),
+            get(self.platform_settings, "default_interfaces.uplink"),
+            [])
+        )
 
     @cached_property
     def uplink_switch_interfaces(self):
-        return get(self._switch_data_combined, "uplink_switch_interfaces")
+        auto_uplink_interfaces = []
+        if self.uplink_switches:
+            # import epdb; epdb.serve()
+            for uplink_switch_index, uplink_switch in enumerate(self.uplink_switches):
+                uplink_switch_facts = get(self._hostvars,
+                                          f"avd_switch_facts..{uplink_switch}..switch", required=False,
+                                          org_key=f"avd_switch_facts.({uplink_switch}).switch",
+                                          separator="..")
+                if len(uplink_switch_facts.default_downlink_interfaces) > 0:
+
+                    actual_parallel_uplinks = int(
+                        len(self.uplink_switches) /
+                        len(list(dict.fromkeys(self.uplink_switches))))
+
+                    # Get the lead interface to each uplink switch
+                    if uplink_switch_index % actual_parallel_uplinks == 0:
```

Comment:
I don't think this will work if the parallel uplinks are not consecutive. (Which they often are not).